src/lists/LList.es6

```diff
@@ -45,11 +55,19 @@ export default class LList {
    */
   prepend(toPre = []) {
     let opts = { 
-      circular  : false,
+      circular  : this.circular,
       prependTo : this.head, 
       oldSize   : this.size,
     };
 
+    //If circular, can't use tail-sharing.
+    if (this.circular){
```

Comment:
input on a clean way to implement this circula duplication functionality, without spinning it into a new function that crowds the api?